File: my-folder/0007-reverse-integer/solution.c
int reverse(int x){
    int result = 0;
    while (x != 0){
		int remainder = x % 10;
		
		if ((result > INT_MAX/10 || result == INT_MAX/10 && remainder > 7) ||
			(result < INT_MIN/10 || result == INT_MIN/10 && remainder < -8))
			return 0;
		
		result = result * 10 + remainder;
		x = x / 10;
    }
    return result;
}

File: my-folder/0007-reverse-integer/solution.rs
impl Solution {
    pub fn reverse(mut x: i32) -> i32 {
        let mut result: i32 = 0;

        while x != 0 {
            let digit = x % 10;
            x /= 10;

            result = match result.checked_mul(10).and_then(|r| r.checked_add(digit)) {
                Some(val) => val,
                None => return 0,
            };
        }

        result
    }
}

